modules/scripts/startup-script/main.tf

```diff
@@ -68,22 +83,30 @@ resource "random_id" "resource_name_suffix" {
   byte_length = 4
 }
 
-resource "google_storage_bucket" "configs_bucket" {
-  project                     = var.project_id
-  name                        = "${var.deployment_name}-startup-scripts-${random_id.resource_name_suffix.hex}"
-  uniform_bucket_level_access = true
+module "config_storage_bucket" {
```

Comment:
using module here so that I can conditionally use it in line #28. I just picked this pattern from existing hpc-toolkit code.

Reply:
A few things I noticed here. 

1. If we use the module (as you currently have), to preserve the existing behavior, we would have to set `force_destroy = true`. Otherwise terraform destroy will fail. 

2. The same `count` syntax works with the `resource`. I personally don't have a reason to prefer the resource to the module except that it would limit the change to a smaller scope. For this reason I might suggest just keeping this as the resource (and adding in the count pattern you have here).

Reply:
Converted to resource in latest iteration and validated it works.